pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -681,6 +678,22 @@ void messageReceived(MessageIdData messageId, ByteBuf headersAndPayload, ClientC
                 }
             });
         }
+
+        if (receiveListener != null) {
```

Comment:
On a second thought, listenerExecutor is already there, and it allows as to serialize the executions of the listener and make sure no two listeners are running concurrently.
Also, on PartitionedConsumer [L122](https://github.com/yahoo/pulsar/pull/83/files#diff-52398d8fd34dbca204933f7cdb615105R122) I'm using executing in listenerExecutor to return immediately.
So maybe it would be better to leave the execution on a listenerThread, one could also share a single listenerThread across topics without any problem